core/include/traccc/seeding/doublet_finding_helper.hpp

```diff
@@ -101,7 +111,12 @@ lin_circle doublet_finding_helper::transform_coordinates(
     scalar iDeltaR2 = 1. / (deltaX * deltaX + deltaY * deltaY);
     scalar iDeltaR = std::sqrt(iDeltaR2);
     //
-    int bottomFactor = 1 * (int(!bottom)) - 1 * (int(bottom));
+    int bottomFactor;
+    if constexpr (otherSpType == details::spacepoint_type::bottom) {
+        bottomFactor = -1;
+    } else {
+        bottomFactor = 1;
+    }
     // cot_theta = (deltaZ/deltaR)
     scalar cot_theta = deltaZ * iDeltaR * bottomFactor;
```

Comment:
I'm not a big fan of this setup... It made sense with the flag arriving at runtime. But if we do it at compilation time, I think this would make more sense:

```c++
scalar cot_theta = deltaZ * iDeltaR;
if constexpr (otherSpType == details::spacepoint_type::bottom) {
    cot_theta = -cot_theta;
}
```